### PR TITLE
Fix all warnings

### PR DESCRIPTION
**Pull request by bmourat**
Added compiler warning flags and fixed most of the warnings.

### CODE REVIEW

File: Classes/BITChannel.m [hunk 1]
```diff
@@ -212,7 +217,7 @@ - (NSString *)serializeDictionaryToJSONString:(NSDictionary *)dictionary {
     BITHockeyLogError(@"ERROR: JSONSerialization error: %@", error.localizedDescription);
     return @"{}";
   } else {
-    return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+    return (NSString *)[[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
```
Comment (MatkovIvan) [inline Classes/BITChannel.m]: Could you explain this one?

File: Classes/BITCrashManager.m [hunk 2]
```diff
@@ -175,36 +177,38 @@ - (NSArray *)callStackReturnAddresses {
 
 
 // C++ Exception Handler
-static void uncaught_cxx_exception_handler(const BITCrashUncaughtCXXExceptionInfo *info) {
+static void uncaught_cxx_exception_handler(const BITCrashUncaughtCXXExceptionInfo *info)  {
```
Comment (MatkovIvan) [inline Classes/BITCrashManager.m]: One space is enough.

File: Classes/BITArrowImageAnnotation.m [hunk 3]
```diff
@@ -31,6 +31,7 @@
 #if HOCKEYSDK_FEATURE_FEEDBACK
 
 #import "BITArrowImageAnnotation.h"
+#import "tgmath.h"
```
Comment (MatkovIvan) [inline Classes/BITArrowImageAnnotation.m]: Usually `<>` is used instead of `""` for system headers.

File: Classes/BITCrashManager.m [hunk 4]
```diff
@@ -142,19 +142,21 @@ static void plcr_post_crash_callback (siginfo_t *info, ucontext_t *uap, void *co
 // This code provides us access to the C++ exception message, including a correct stack trace.
 //
 @interface BITCrashCXXExceptionWrapperException : NSException
+
 - (instancetype)initWithCXXExceptionInfo:(const BITCrashUncaughtCXXExceptionInfo *)info;
+
+@property (nonatomic, readonly) BITCrashUncaughtCXXExceptionInfo *info;
```
Comment (MatkovIvan) [inline Classes/BITCrashManager.m]: Missing `const` modifier

File: Classes/BITCrashManager.m (hunk 2 above)
Comment (MatkovIvan) [inline Classes/BITCrashManager.m]: ` __attribute__((noreturn))`?

**Comment by bmourat** [comment]
There are couple of warnings left but you can only get rid of them by specifying different -no flags, but i don't want to add this flags as this warnings can be helpful in future.

**Comment by TroubleMakerBen** [comment]
@bmourat merging this. We'll get rid of more warnings over time. =)

**Pull request merged**